### ISSUE TITLE
schema_key::operator== is inconsistent with std::hash<schema_key> specialisation

### ISSUE THREAD

**Issue by davemcincork** [microsoft/krabsetw#139]
This issue manifests in my application as a memory leak in the schema_locator:
- The ``schema_key::operator==`` overload does a member-wise comparison of all five fields in the struct.
- The ``std::hash<schema_key>`` specialisation acts upon all bytes in the struct, including the padding bytes that are not evaluated by ``schema_key::operator==``.
- The result is that two keys which compare equal will not necessarily hash equal.
- This is a problem because for std::unordered_map to work correctly, it is necessary that ``key1 == key2`` => ``hash(key1) == hash(key2)``.
- And this causes the schema_locator to have duplicate entries for the same event ID.

The following test program illustrates the problem:
```C++
#define WIN32_LEAN_AND_MEAN
#include <krabs.hpp>

int main()
{
    const EVENT_RECORD eventRecord; // uninitialised garbage but that's OK

    const krabs::schema_key key1{ eventRecord };
    const krabs::schema_key key2{ eventRecord };

    const std::hash<krabs::schema_key> hash;

    auto exitCode = EXIT_SUCCESS;

    if ((key1 == key2) != (hash(key1) == hash(key2)))
    {
        fprintf(stderr, "Oops!\n");
        exitCode = EXIT_FAILURE;
    }

    return exitCode;
}
```
A Release build done with VS2019.latest will output "Oops!". A Debug build will not because the stack is initialised.

**Comment by davemcincork** [comment]
I forgot to mention that the issue is fixed by using ``#pragma pack`` to set the schema_key struct packing level to 1, thereby removing the packing bytes and working around the broken hash function.

**Comment by swannman** [comment]
@davemcincork thanks for reporting this and for providing a minimal repro! I've submitted a PR to fix: #140

**Comment by davemcincork** [comment]
Many thanks for fixing so quickly. Sorry I couldn’t do so myself...legal ...

**Comment by swannman** [comment]
This is now fixed in nuget v4.1.14. Cheers!
